xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -98,6 +101,18 @@ bool CDVDInputStreamNavigator::Open()
 #endif
 
   // open up the DVD device
+  if (m_item.IsDiscImage())
+  {
+    // if dvd image file (ISO or alike) open using libdvdnav stream callback functions
+    m_pstream = new CDVDInputStreamFile(m_item);
+    if (!m_pstream->Open() || m_dll.dvdnav_open_stream(&m_dvdnav, m_pstream, &m_dvdnav_stream_cb) != DVDNAV_STATUS_OK)
```

Comment:
use `m_pstream.get()` here to avoid passing the unique_ptr as it makes the casts easier in the static functions.